handler/producer/result.go

```diff
@@ -79,7 +89,7 @@ func roundtrip(rt http.RoundTripper, req *http.Request, results chan<- *Result,
 		if !seen {
 			results <- &Result{
 				Beresp:        beresp,
-				Err:           errors.UnexpectedStatus.With(err),
+				Err:           errors.UnexpectedStatus.With(err).Status(http.StatusBadGateway),
```

Comment:
It is not clear why this change is necessary. `errors.UnexpectedStatus` is a sub-type of `errors.Endpoint`, which already has `http.StatusBadGateway`.